XenAdmin/Controls/ComboBoxes/VgpuComboBox.cs

```diff
@@ -195,6 +195,9 @@ public bool Equals(GpuTuple other)
                     if (!VgpuTypes[i].Equals(other.VgpuTypes[i]))
                         return false;
                 }
+
+                if (IsGpuHeaderItem && other.IsVgpuSubitem)

```

Comment:
In order to make a valid Equals function, we need (a == b) => (b == a). I.e., we need the reverse case here as well: this.IsVgpuSubitem && other.IsGpuHeaderItem.